airflow/providers/amazon/aws/operators/ecs.py

```diff
@@ -260,6 +265,24 @@ def _wait_for_task_ended(self) -> None:
         waiter.config.max_attempts = sys.maxsize  # timeout is managed by airflow
         waiter.wait(cluster=self.cluster, tasks=[self.arn])
 
+    def _cloudwatch_log_events(self):
+        if not self.awslogs_group:
+            return
+
+        if not self.awslogs_stream_prefix:
+            return
+
+        task_id = self.arn.split("/")[-1]
+        stream_name = "{}/{}".format(self.awslogs_stream_prefix, task_id)
+
+        return self.get_logs_hook().get_log_events(self.awslogs_group, stream_name)
+
+    def _last_log_event(self):
+        log_events = self._cloudwatch_log_events()
+        if log_events is not None:
+            return deque(log_events, maxlen=1).pop()["message"]
```

Comment:
May you please clarify why `deque` is needed here? Thanks.

Reply:
It's a more efficient way to get the last item from a generator - no need to iterate over every item from the generator just to get to the last one.

It might make a difference in case of lots of events.